hermes-consumers/src/main/java/pl/allegro/tech/hermes/consumers/supervisor/workload/SubscriptionAssignmentView.java

```diff
@@ -0,0 +1,204 @@
+package pl.allegro.tech.hermes.consumers.supervisor.workload;
+
+import com.google.common.collect.ImmutableSet;
+import com.google.common.collect.Sets;
+import pl.allegro.tech.hermes.api.SubscriptionName;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.function.BiConsumer;
+
+import static java.util.stream.Collectors.toList;
+import static java.util.stream.Collectors.toSet;
+
+public class SubscriptionAssignmentView {
```

Comment:
Well the primary idea behind *View* suffix was that it is an immutable view of underlying zk structure... There is some transforming functionality added which might challenge the name a little bit but the view itself represents snapshot of zk structure...